agave/filters.py

```diff
@@ -20,3 +23,68 @@ def generic_query(query: QueryParams) -> Q:
     if 'count' in fields:
         del fields['count']
     return filters & Q(**fields)
+
+
+def get_by_id(cls, id: str):
+    try:
+        id_obj = cls.model.objects.get(id=id)
+    except Exception as exc:
+        if isinstance(exc, DoesNotExist):
+            raise
+        id_obj = cls.model.get_by(id=id)
+        if not id_obj:
+            raise
+    return id_obj
+
+
+def get_by_id_and_user(cls, id: str):
```

Comment:
you should accept `user_id` as a parameter to make it easier to test

Reply:
Done!